### ISSUE TITLE
Missing Version Warning

### ISSUE THREAD

**Issue by marens101** [squidfunk/mkdocs-material#3013]
### Contribution guidelines

- [X] I've read the [contribution guidelines](https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md) and wholeheartedly agree

### I've found a bug and checked that ...

- [X] ... the problem doesn't occur with the `mkdocs` or `readthedocs` themes
- [ ] ... the problem persists when all overrides are removed, i.e. `custom_dir`, `extra_javascript` and `extra_css`
- [X] ... the documentation does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem

### Description

[Version Warning](https://squidfunk.github.io/mkdocs-material/setup/setting-up-versioning/#version-warning) banner does not show on any pages when an updated version is deployed.

### Expected behaviour

A version warning banner is shown on outdated versions of the documentation

### Actual behaviour

No banner is shown

### Steps to reproduce

Mike 1.0.1 is being used to track versions. 
We decided to add the version warning banner when releasing a new version of the documentation, so added the following to the file `overrides/main.html` as detailed [here](https://squidfunk.github.io/mkdocs-material/customization/#overriding-blocks-recommended) and [here](https://squidfunk.github.io/mkdocs-material/setup/setting-up-versioning/#version-warning).
```
{% extends "base.html" %}

{% block outdated %}
  You're not viewing the latest version.
  <a href="{{ '../' ~ base_url }}"> 
    <strong>Click here to go to latest.</strong>
  </a>
{% endblock %}
```

When using Mike to build serve a new version of the docs locally (lets call it version 2), the banner is not shown. I assumed this was due to Mike only building the latest docs version and leaving older versions untouched, causing version 1 to not have the banner. I deployed again, this time as version 3, however the banner still does not show on either versions 1 or 2. I am updating the latest tag using Mike, and the `mike list` command confirms this.

I then thought that maybe I misconfigured the overrides, however testing with the `announce` block proves that this is not the case.

### Package versions

- Python: 3.9.1
- MkDocs: 1.2.2
- Material: 7.2.6+insiders.2.13.3

### Configuration

```yaml
site_name: Utils Manual
site_dir: public
remote_branch: deploy
theme:
  name: material
  custom_dir: overrides
```

### System information

- Operating system: Windows 11
- Browser: Chrome 93

**Comment by squidfunk** [comment]
Thanks for reporting. Do you have your site deployed somewhere, so I could check what's wrong?

**Comment by marens101** [comment]
I don't currently have a version running with the block enabled- after finding it wasn't working locally I decided to come back to it later when it was fixed and just shoved it onto another branch.

If it helps though, I could spin up a clone of the live site with the block

**Comment by squidfunk** [comment]
I have a working version of this running here:
https://squidfunk.github.io/mkdocs-material-example-versioning/

Although it uses a slightly older version of Insiders, it should still work the same, as nothing was changed on the version selector in later versions. There still might be some unknown regression, though. A deployed example would make debugging much easier.

**Comment by marens101** [comment]
[sampledocs.zip](https://github.com/squidfunk/mkdocs-material/files/7140814/sampledocs.zip)
[sampledocs-source.zip](https://github.com/squidfunk/mkdocs-material/files/7140838/sampledocs-source.zip)
Attached are some sample docs, both the deployed version generated with `mike deploy` and the source and config used to create it.

**Comment by squidfunk** [comment]
Thanks for the sample code. I've checked it and see no apparent error. Additionally, I've just verified that the feature works, and it does, so as expected there was no regression that broke it. See:

- https://squidfunk.github.io/version-test/0.1/
- https://squidfunk.github.io/version-test/0.2/

For this reason, the error must be somewhere in your setup. The general advice is to follow the [versioning example](https://github.com/squidfunk/mkdocs-material-example-versioning).

The test linked above was deployed with:

- mkdocs-1.2.2
- mkdocs-material-7.2.6+insiders-2.13.3
- mike-1.1.0
